src/fake/mod.rs

```diff
@@ -222,15 +231,59 @@ impl FileSystem for FakeFileSystem {
     }
 
     fn readonly<P: AsRef<Path>>(&self, path: P) -> Result<bool> {
-        self.apply(path.as_ref(), |r, p| r.readonly(p))
+        apply(&self.registry, path.as_ref(), |r, p| r.readonly(p))
     }
 
     fn set_readonly<P: AsRef<Path>>(&self, path: P, readonly: bool) -> Result<()> {
         self.apply_mut(path.as_ref(), |r, p| r.set_readonly(p, readonly))
     }
 
     fn len<P: AsRef<Path>>(&self, path: P) -> u64 {
-        self.apply(path.as_ref(), |r, p| r.len(p))
+        apply(&self.registry, path.as_ref(), |r, p| r.len(p))
+    }
+}
+
+#[derive(Debug)]
+pub struct FakeOpenFile {
+    registry: Arc<Mutex<Registry>>,
```

Comment:
I'm very open to this, and it would be easy to do. If an `OpenFile` tries to access the registry after it has been dropped, we could return an `io::Error(ErrorKind::Other)`. Because there is no equivalent on a real fs, other than yanking the disk from a running system :)

Curious about what the use case would be?